tools/isobuild/isopack.js

```diff
@@ -44,215 +44,134 @@ var rejectBadPath = function (p) {
-  var isopackJsonPath = files.pathJoin(isopackDirectory, "isopack.json");
-  var unipackageJsonPath = files.pathJoin(isopackDirectory, "unipackage.json");
-
-  if (files.exists(isopackJsonPath)) {
-    var isopackJson = JSON.parse(files.readFile(isopackJsonPath));
-
-    if (isopackJson['isopack-2']) {
-      metadata = isopackJson['isopack-2'];
-      originalVersion = 'isopack-2';
-    } else if (isopackJson['isopack-1']) {
-      metadata = Isopack.convertIsopackFormat(
-        isopackJson['isopack-1'], 'isopack-1', 'isopack-2');
-      originalVersion = 'isopack-1';
-    } else {
-      // This file is from the future and no longer supports this version
-      throw new Error("Could not find isopack data supported any supported format (isopack-1 or isopack-2).\n" +
-        "This isopack was likely built with a much newer version of Meteor.");
-    }
-  } else if (files.exists(unipackageJsonPath)) {
-    // super old version with different file name
+  // These functions are designed to convert isopack metadata between
+  // versions. They were designed to convert between unipackage-pre2 and
+  // isopack-1. The differences between these formats are essentially syntactical,
+  // not semantic, and occur entirely in the isopack.json file, not in the
+  // individual unibuild json files. These functions are written assuming those
+  // constraints, and were not actually useful in the isopack-1/isopack-2
+  // transition,where most of the changes are in the unibuild level, and there's
+  // actual semantic changes involved. So they are not actually used as much as
+  // they were before.
+  static convertOneStepForward(data, fromFormat) {
+    var convertedData = _.clone(data);
     // XXX COMPAT WITH 0.9.3
-    if (files.exists(unipackageJsonPath)) {
-      metadata = JSON.parse(files.readFile(unipackageJsonPath));
-
-      metadata = Isopack.convertIsopackFormat(metadata,
-        "unipackage-pre2", "isopack-2");
-      originalVersion = 'unipackage-pre2';
+    if (fromFormat === "unipackage-pre2") {
+      convertedData.builds = convertedData.unibuilds;
+      delete convertedData.unibuilds;
+      return convertedData;
     }
-
-    if (metadata.format !== "unipackage-pre2") {
-      // We don't support pre-0.9.0 isopacks, but we do know enough to delete
-      // them if we find them in an isopack cache somehow (rather than crash).
-      if (metadata.format === "unipackage-pre1") {
-        throw new exports.OldIsopackFormatError();
-      }
-
-      throw new Error("Unsupported isopack format: " +
-                      JSON.stringify(metadata.format));
+    if (fromFormat === "isopack-1") {
+      // For now, there's no difference in this direction at the isopack level.
+      return convertedData;
     }
   }
 
-  return {metadata, originalVersion};
-});
+  static convertOneStepBackward(data, fromFormat) {
+    var convertedData = _.clone(data);
+    if (fromFormat === "isopack-1") {
```

Comment:
Not to be curdmudgeony, but is there an enum or exported const for this? :L

Reply:
This is the kind of thing that will be much easier to improve once everything is TypeScript. Remember that `enum` is a TypeScript-specific feature, and we have only just begun the conversion to TypeScript. I think @Slava made the right call preserving the existing code style in this PR.